src/stores/DialogStore/DialogStore.ts

```diff
@@ -20,179 +62,78 @@ export class DialogStore {
         return DialogStore.staticInstance;
     }
 
-    // Region
-    @observable regionDialogVisible: boolean;
-    @action showRegionDialog = () => {
-        this.regionDialogVisible = true;
-    };
-    @action hideRegionDialog = () => {
-        this.regionDialogVisible = false;
-    };
-
-    // Hotkey
-    @observable hotkeyDialogVisible: boolean;
-    @action showHotkeyDialog = () => {
-        this.hotkeyDialogVisible = true;
-    };
-    @action hideHotkeyDialog = () => {
-        this.hotkeyDialogVisible = false;
-    };
-
-    // About
-    @observable aboutDialogVisible: boolean;
-    @action showAboutDialog = () => {
-        this.aboutDialogVisible = true;
-    };
-    @action hideAboutDialog = () => {
-        this.aboutDialogVisible = false;
-    };
-
-    // Preference
-    @observable preferenceDialogVisible: boolean;
-    @action showPreferenceDialog = () => {
-        this.preferenceDialogVisible = true;
-    };
-    @action hidePreferenceDialog = () => {
-        this.preferenceDialogVisible = false;
-    };
-
-    // Layout
-    @observable saveLayoutDialogVisible: boolean;
-    @action showSaveLayoutDialog = (oldLayoutName?: string) => {
-        this.saveLayoutDialogVisible = true;
-        AppStore.Instance.layoutStore.setOldLayoutName(oldLayoutName);
-    };
-    @action hideSaveLayoutDialog = () => {
-        this.saveLayoutDialogVisible = false;
-    };
-
-    // Workspace
-    @observable workspaceDialogMode = WorkspaceDialogMode.Hidden;
-    @action showWorkspaceDialog = (mode = WorkspaceDialogMode.Save) => {
-        this.fileBrowserDialogVisible = false;
-        this.workspaceDialogMode = mode;
-    };
-    @action hideWorkspaceDialog = () => {
-        this.workspaceDialogMode = WorkspaceDialogMode.Hidden;
-    };
-
-    // Workspace sharing
-    @observable shareWorkspaceDialogVisible: boolean;
-    @action showShareWorkspaceDialog = () => {
-        this.shareWorkspaceDialogVisible = true;
-    };
-    @action hideShareWorkspaceDialog = () => {
-        this.shareWorkspaceDialogVisible = false;
+    zIndexManager = AppStore.Instance.zIndexManager;
+
+    @action showDialog = (id: string, props?: showDialogProps) => {
```

Comment:
I suggest we rename `props` to `options`.

Reply:
okay.